cadence/contracts/auth/AxelarAuthWeighted.cdc

```diff
@@ -0,0 +1,189 @@
+import Crypto
+
+// Main Auth Weighted contract for audit
+pub contract AxelarAuthWeighted {
+  pub var currentEpoch: UInt256
+  pub let hashForEpoch: {UInt256: String}
+  pub let epochForHash: {String: UInt256}
+
+  priv let OLD_KEY_RETENTION: UInt256
+
+  pub event OperatorshipTransferred(
+    newOperators: [String],
+    newWeights: [UInt256],
+    newThreshold: UInt256
+  )
+
+  pub struct TransferOperatorshipParams {
+    pub let newOperators: [String]
+    pub let newWeights: [UInt256]
+    pub let newThreshold: UInt256
+
+    init(newOperators: [String], newWeights: [UInt256], newThreshold: UInt256) {
+        self.newOperators = newOperators
+        self.newThreshold = newThreshold
+        self.newWeights = newWeights
+    }
+  }
+
+  /**************************\
+  |* External Functionality *|
+  \**************************/
+  pub fun validateProof(message: String, operators: [String], weights: [UInt256], threshold: UInt256, signatures: [String]): Bool {
+    let operatorsHash = self._operatorsToHash(operators: operators, weights: weights, threshold: threshold)
+    let operatorsEpoch = self.epochForHash[operatorsHash]
+    let epoch = self.currentEpoch
+
+    if operatorsEpoch == nil || epoch - operatorsEpoch! >= self.OLD_KEY_RETENTION {
+      panic("Invalid operators")
+    }
+
+    self._validateSignatures(message: message, operators: operators, weights: weights, threshold: threshold, signatures: signatures)
+
+    return operatorsEpoch! == epoch
+  }
+
+  /***********************\
+  |* Owner Functionality *|
+  \***********************/
+  access(account) fun transferOperatorship(params: TransferOperatorshipParams) {
```

Comment:
This pattern makes sense given the call in AxelarGateway and matches the solidity implementation, but it has me wondering if there are plans for the governance of this contract account? I notice that neither called method includes input validation, so whoever owns the account can transfer operatorship without the need for valid signatures or meeting a threshold to do so.